showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java

```diff
@@ -0,0 +1,256 @@
+            .build();
+    Policy policy = grpcClient.setIamPolicy(policyRequest);
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testHttpJson_setIamPolicy() {
+    SetIamPolicyRequest policyRequest =
+        SetIamPolicyRequest.newBuilder()
+            .setPolicy(DEFAULT_POLICY)
+            .setResource(resourceName)
+            .build();
+    Policy policy = httpjsonClient.setIamPolicy(policyRequest);
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testGrpc_setIamPolicy_missingResource() {
+    SetIamPolicyRequest setIamPolicyRequest = SetIamPolicyRequest.newBuilder().build();
+    assertThrows(
+        InvalidArgumentException.class, () -> grpcClient.setIamPolicy(setIamPolicyRequest));
+  }
+
+  @Test
+  public void testHttpJson_setIamPolicy_missingResource() {
+    SetIamPolicyRequest setIamPolicyRequest = SetIamPolicyRequest.newBuilder().build();
+    assertThrows(
+        InvalidArgumentException.class, () -> httpjsonClient.setIamPolicy(setIamPolicyRequest));
+  }
+
+  @Test
+  public void testGrpc_setIamPolicy_missingPolicy() {
+    SetIamPolicyRequest setIamPolicyRequest =
+        SetIamPolicyRequest.newBuilder().setResource(resourceName).build();
+    assertThrows(
+        InvalidArgumentException.class, () -> grpcClient.setIamPolicy(setIamPolicyRequest));
+  }
+
+  @Test
+  public void testHttpJson_setIamPolicy_missingPolicy() {
+    SetIamPolicyRequest setIamPolicyRequest =
+        SetIamPolicyRequest.newBuilder().setResource(resourceName).build();
+    assertThrows(
+        InvalidArgumentException.class, () -> httpjsonClient.setIamPolicy(setIamPolicyRequest));
+  }
+
+  @Test
+  public void testGrpc_getIamPolicy() {
+    SetIamPolicyRequest policyRequest =
+        SetIamPolicyRequest.newBuilder()
+            .setPolicy(DEFAULT_POLICY)
+            .setResource(resourceName)
+            .build();
+    grpcClient.setIamPolicy(policyRequest);
+
+    Policy policy =
+        grpcClient.getIamPolicy(
+            GetIamPolicyRequest.newBuilder().setResource(policyRequest.getResource()).build());
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testHttpJson_getIamPolicy() {
+    SetIamPolicyRequest policyRequest =
+        SetIamPolicyRequest.newBuilder()
+            .setPolicy(DEFAULT_POLICY)
+            .setResource(resourceName)
+            .build();
+    httpjsonClient.setIamPolicy(policyRequest);
+
+    Policy policy =
+        httpjsonClient.getIamPolicy(
+            GetIamPolicyRequest.newBuilder().setResource(policyRequest.getResource()).build());
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testGrpc_getIamPolicy_doesNotExist() {
+    GetIamPolicyRequest getIamPolicyRequest =
+        GetIamPolicyRequest.newBuilder().setResource(resourceName).build();
+    assertThrows(NotFoundException.class, () -> grpcClient.getIamPolicy(getIamPolicyRequest));
```

Comment:
I don't think we want to test this scenario, as this exception is based on a backend response. For example, not likely but the backend could return a different status code which will be mapped to a different exception.

Reply:
This was mapped from the showcase server's error responses: https://github.com/googleapis/gapic-showcase/blob/b146f892d98ea0d733c45a2315941e8e6ab69c90/server/services/iam_policy_service.go#L27-L29

Reply:
If the error is from the backend, I don't think we want to test them, or at most test one of them to make sure exceptions from IAM server can be properly mapped. Test all the permutations is not needed.

Reply:
Sounds good. I'll keep one test and remove the rest.